syntax = "proto3";
package protos.playbooks;

import "google/protobuf/wrappers.proto";


message Cloudwatch {
  message EcsListClusters {
    // No parameters needed for listing all clusters
  }
  
  message EcsListTasks {
    google.protobuf.StringValue cluster_name = 1;
  }
  
  message EcsGetTaskLogs {
    google.protobuf.StringValue cluster_name = 1;
    google.protobuf.StringValue task_definition = 2;
    google.protobuf.Int64Value max_lines = 3;
  }
  message MetricExecution {
    message Dimension {
      google.protobuf.StringValue name = 1;
      google.protobuf.StringValue value = 2;
    }
    google.protobuf.StringValue namespace = 1;
    google.protobuf.StringValue region = 2;
    google.protobuf.StringValue metric_name = 3;
    repeated Dimension dimensions = 4;
    google.protobuf.StringValue statistic = 5;
    repeated uint32 timeseries_offsets = 6;
    google.protobuf.UInt64Value period = 7;
  }

  message FilterLogEvents {
    google.protobuf.StringValue region = 1;
    google.protobuf.StringValue log_group_name = 2;
    google.protobuf.StringValue filter_query = 3;
  }
  message RdsGetSqlQueryPerformanceStats {
    google.protobuf.StringValue db_resource_uri = 1;
  }

  message FetchDashboard {
    google.protobuf.StringValue dashboard_name = 1;
    google.protobuf.Int32Value step = 2;
  }

  message FetchS3File {
    google.protobuf.StringValue bucket_name = 1;
    google.protobuf.StringValue object_key = 2;
  }


  enum TaskType {
    UNKNOWN = 0;
    METRIC_EXECUTION = 1;
    FILTER_LOG_EVENTS = 2;
    RDS_GET_SQL_QUERY_PERFORMANCE_STATS = 3;
    ECS_LIST_CLUSTERS = 4;
    ECS_LIST_TASKS = 5;
    ECS_GET_TASK_LOGS = 6;
    FETCH_DASHBOARD = 7;
    FETCH_S3_FILE = 8;
  }

  TaskType type = 1;
  oneof task {
    MetricExecution metric_execution = 2;
    FilterLogEvents filter_log_events = 3;
    RdsGetSqlQueryPerformanceStats rds_get_sql_query_performance_stats = 4;
    EcsListClusters ecs_list_clusters = 5;
    EcsListTasks ecs_list_tasks = 6;
    EcsGetTaskLogs ecs_get_task_logs = 7;
    FetchDashboard fetch_dashboard = 8;
    FetchS3File fetch_s3_file = 9;
  }
}